decision_service/rlclientlib/bg_model_download.cc

```diff
@@ -0,0 +1,32 @@
+#include "bg_model_download.h"
+#include "api_status.h"
+
+namespace reinforcement_learning { namespace model_management {
+  bg_model_download::bg_model_download(i_data_transport* ptrans, data_callback_fn* pdata_cb)
+    : _ptrans(ptrans), _pdata_cb(pdata_cb){}
+
+  bg_model_download::~bg_model_download() { delete _ptrans; }
+
+  bg_model_download::bg_model_download(bg_model_download&& temp) noexcept {
+    _ptrans = temp._ptrans;
+    temp._ptrans = nullptr;
+    _pdata_cb = temp._pdata_cb;
+    temp._pdata_cb = nullptr;
+  }
+
+  bg_model_download& bg_model_download::operator=(bg_model_download&& temp) noexcept {
+    if (&temp != this) {
+      const auto x = _ptrans;
+      _ptrans      = temp._ptrans;
+      temp._ptrans = x;
+    }
+    return *this;
+  }
```

Comment:
are those used or just for good measure?

Reply:
I might have used it in some tests at some point.  

---
In reply to: [193485495](https://github.com/JohnLangford/vowpal_wabbit/pull/1495#discussion_r193485495) [](ancestors = 193485495)